File: src/test/scala/com/yammer/dropwizard/examples/SayCommand.scala
package com.yammer.dropwizard.examples

import com.yammer.dropwizard.cli.{Flag, ConfiguredCommand}
import com.yammer.dropwizard.Service
import com.codahale.fig.Configuration

class SayCommand extends ConfiguredCommand {
  def name = "say"

  override def description = Some("Prints out the saying to console")

  override def options = Flag("v", "verbose", "yell it a lot") :: Nil
  
  def run(service: Service, config: Configuration, opts: Map[String, List[String]], args: List[String]) = {
    val saying = SayingFactory.buildSaying(config)
    for (i <- 1 to (if (opts.contains("verbose")) 10 else 1)) {
      log.warn(saying)
    }
    None
  }
}

File: dropwizard/src/test/java/com/yammer/flopwizard/example/SayCommand.java
package com.yammer.flopwizard.example;

import com.yammer.dropwizard.Service;
import com.yammer.dropwizard.cli.ConfiguredCommand;
import com.yammer.dropwizard.config.Configuration;
import org.apache.commons.cli.CommandLine;
import org.apache.commons.cli.Options;

@SuppressWarnings("NullableProblems")
public class SayCommand extends ConfiguredCommand<ExampleConfiguration> {
    public SayCommand() {
        super(ExampleConfiguration.class, "say", "Prints out the saying to console");
    }

    @Override
    public Options getOptions() {
        return new Options().addOption("v", "verbose", false, "yell it a lot");
    }

    @Override
    protected void run(Service service,
                       ExampleConfiguration configuration,
                       CommandLine params) throws Exception {
        final String saying = configuration.getSaying();
        int max = 1;
        if (params.hasOption("verbose")) {
            max = 10;
        }

        for (int i = 0; i < max; i++) {
            System.err.println(saying);
        }
    }
}
